build.rs

```diff
@@ -73,15 +74,18 @@ impl Version {
         }
         let minor = try!(num.parse::<u32>().map_err(|e| e.to_string()));
 
+        let mut stable = true;
         num.clear();
         for c in parts[2].chars() {
             if !c.is_digit(10) {
+                // last version part ends with "-nightly" or "-beta"
+                stable = false;
```

Comment:
Have you tested this? It seems to me like this will always set `stable = false;`.

Reply:
Ops, you are right. I tested on the problematic nightly only.